src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -7140,7 +7140,7 @@ public PropertyNameCompleter(string parameterNameOfInput)
             CommandAst commandAst,
             IDictionary fakeBoundParameters)
         {
-            if (!(commandAst.Parent is PipelineAst pipelineAst))
+            if (commandAst.Parent is not PipelineAst pipelineAst)
```

Comment:
I wonder if the var `pipelineAst` will be populated when `commandAst.Parent` is `PipelineAst`. Would love to confirm it.

Reply:
I was wondering the same, but given it's still essentially equivalent to the old syntax and still supports pattern matching, I'm reasonably confident it should be 😂

Reply:
You can show `pipelineAst` is populated by enabling nullable analysis for `PropertyNameCompleter(string parameterNameOfInput)`

https://github.com/PowerShell/PowerShell/blob/bf0f1ea7742c8853fd1cd2d3896b7f353aaf3e67/src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs#L7149
Compiler says `pipelineAst` is not null here.